src/DynamoCore/Extensions/ExtensionManager.cs

```diff
@@ -23,6 +57,29 @@ public class ExtensionManager: IExtensionManager, ILogSource
         public ExtensionManager()
         {
             extensionLoader.MessageLogged += Log;
+            this.ExtensionAdded += AddEventSubscriptions;
+            this.ExtensionRemoved += removeEventSubscriptions;
+        }
+
+        private void removeEventSubscriptions(IExtension obj)
+        {
+            if (obj is IExtensionSource)
+            {
+                (obj as IExtensionSource).RequestLoadExtensionHandler -= this.ExtensionLoader.Load;
+                (obj as IExtensionSource).RequestAddExtensionHandler -= this.Add;
+            }
+        }
+
+        private void AddEventSubscriptions(IExtension obj)
```

Comment:
`SubscribeExtension` & `UnsubscribeExtension` is probably a more evocative name.